packages/dd-trace/src/config.js

```diff
@@ -321,28 +319,31 @@ class Config {
     this.#defaults = defaults
     this.#applyDefaults()
     this.#applyStableConfig(this.stableConfig?.localEntries ?? {}, this.#localStableConfig)
-    this.#applyEnvironment(envs)
+    this.#applyEnvironment()
     this.#applyStableConfig(this.stableConfig?.fleetEntries ?? {}, this.#fleetStableConfig)
     this.#applyOptions(options)
     this.#applyCalculated()
     this.#applyRemote({})
     this.#merge()
 
-    tagger.add(this.tags, {
-      service: this.service,
-      env: this.env,
-      version: this.version,
+    // Use typed version of `this` to appease the type checker
+    const self = /** @type {Record<string, any>} */ (this)
```

Comment:
I don't think we should do that. In fact, we should prohibit `any` usage in general.

This is just hiding the underlying issue of the types not being detected properly. So let us tackle that instead.

Reply:
Reverted